src/toil/jobStores/awsJobStore.py

```diff
@@ -460,35 +504,45 @@ def _newFileID(self, sharedFileName=None):
         else:
             return str(uuid.uuid5(self.sharedFileJobID, str(sharedFileName)))
 
-    def _getFileVersionAndBucket(self, jobStoreFileID):
+    def _getFile(self, jobStoreFileID, isProtected=True):
         """
+        new object that is File or Version
         :rtype: tuple(str version, AWS bucket)
         """
         item = None
         for attempt in retry_sdb():
             with attempt:
+                #what attributes does this table have
                 item = self.versions.get_attributes(item_name=jobStoreFileID,
-                                                    attribute_name=['version', 'bucketName'],
                                                     consistent_read=True)
-        assert item is not None
-        bucketName = item.get('bucketName')
-        if bucketName is None:
-            return None, None
+        fileType = item.get('fileType', None)
+        if fileType is None:
+            # this is only possible if register was not calledo
+            return None
         else:
-            # noinspection PyTypeChecker
-            return item.get('version'), getattr(self, bucketName)
+            file = File(self,jobStoreFileID,fileType)
+            try:
+                file.version=item['version']
+            except KeyError:
+                file.buffer=self._fromSDBItem(item, isProtected)
+            return file
 
     def _getFileVersion(self, jobStoreFileID, expectedBucket=None):
-        version, bucket = self._getFileVersionAndBucket(jobStoreFileID)
-        if bucket is None:
-            assert version is None
+        file = self._getFile(jobStoreFileID)
+        if file is None or file.bucket is None:
```

Comment:
It should be getFile()'s job to ensure that it doesn't return garbage. Whatever is being validated here should be validate in getFile().